packages/compiler-cli/src/ngtsc/docs/src/initializer_api_function_extractor.ts

```diff
@@ -213,17 +216,20 @@ function extractFunctionWithOverloads(
 ): FunctionWithOverloads {
   return {
     name,
-    signatures: filterSignatureDeclarations(signatures).map((s) => ({
+    signatures: filterSignatureDeclarations(signatures).map(({decl, signature}) => ({
       name,
       entryType: EntryType.Function,
-      description: extractJsDocDescription(s),
-      generics: extractGenerics(s),
+      description: extractJsDocDescription(decl),
+      generics: extractGenerics(decl),
       isNewType: false,
-      jsdocTags: extractJsDocTags(s),
-      params: extractAllParams(s.parameters, typeChecker),
-      rawComment: extractRawJsDoc(s),
+      jsdocTags: extractJsDocTags(decl),
+      params: extractAllParams(decl.parameters, typeChecker),
+      rawComment: extractRawJsDoc(decl),
       returnType: typeChecker.typeToString(
-        typeChecker.getReturnTypeOfSignature(typeChecker.getSignatureFromDeclaration(s)!),
+        typeChecker.getReturnTypeOfSignature(signature),
+        undefined,
+        // This ensures that e.g. `T | undefined` is reduced to `T`.
```

Comment:
nit: this seems inverted

```suggestion
        // This ensures that e.g. `T | undefined` is not reduced to `T`.
```

Reply:
Fixed!